### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by 5hayanB**
added rvfi tracer module

### CODE REVIEW

File: src/main/scala/components/Core.scala [hunk 1]
```diff
@@ -3,9 +3,9 @@ package nucleusrv.components
 import chisel3._
 import chisel3.util._
 
-class Core(M:Boolean = false) extends Module {
+class Core(M:Boolean = false, RVFI:Boolean=false, XLEN:Int) extends Module {
```
Comment (Talha-Ahmed-1) [inline src/main/scala/components/Core.scala:6]: write default value XLEN it should compile in case user didn't provide it in instance.

Comment (CyberGai) [reply src/main/scala/components/Core.scala:6]: ok

**Comment by Talha-Ahmed-1** [comment]
Run compliance tests on it.

**Comment by Talha-Ahmed-1** [comment]
@5hayanB all the compliance of I M and C are passing now ?

**Comment by 5hayanB** [comment]
J instruction is failing, taking a look at it right now

**Pull request closed without merging**